### PR TITLE
Version 2.0.0-dev.3.7

### PR DESCRIPTION
**Pull request by lexaknyazev**
### New Features

* Added `GLB_EXTRA_DATA` warning. Extra bytes beyond the declared GLB total length are now ignored.

### Bugfixes

* Fixed crash when exceeding the maximum reported issues threshold while parsing GLB structures (#180).

* Fixed erroneous line break in the generated issues table (#174).

### Integration updates

* ZIP-archives generated on Windows can now be opened on other platforms.

### CODE REVIEW

File: tool/grind.dart [hunk 1]
```diff
@@ -61,7 +62,7 @@ void issues() {
       } on TypeError catch (_) {
         message = issueType.message(argsWithArray);
       }
-      sb.writeln('|${issueType.code}|$message|'
+      sb.writeln('|${issueType.code}|${message.replaceAll('\n', ' ')}|'
```
Comment (javagl) [inline tool/grind.dart:65]: I wondered about the options of 

- removing the `\n` here
- omitting it at https://github.com/KhronosGroup/glTF-Validator/blob/9ecda14d6d2899d3680182ca93c6aea95a1662c3/lib/src/errors.dart#L248 in the first place

(The "Parser output" info from the `invalidJson` does not have the `\n`...). 
But don't have any basis to say what would be "better". I could imagine that, depending on the output, the `\n` adds clarity or readability.

**Review by javagl — APPROVED** [review]
I tried to reproduce issue https://github.com/KhronosGroup/glTF-Validator/issues/180 , but this apparently _only_ happens in the web version? I'm not sure about the translation process from Dart to the minified JavaScript. The "desktop version" (before the fix) seems to work and create
```
{
    "uri": "test/base/data/glb/todo.glb",
    "mimeType": "model/gltf-binary",
    "validatorVersion": "2.0.0-dev.3.5",
    "issues": {
        "numErrors": 22310,
        "numWarnings": 22333,
...
```
with the input file that was given as a test case. (Speaking of test cases: I wondered whether there should be some `glb-with-lots-of-zeros.glb` as "unit test" for that, but maybe the `extra_data.glb` sufficiently covers this)

I don't know what happens in `grind.dart` in all detail, but at least can confirm that with the changes done there, `dart run grinder exe-archive` (still) works.

File: tool/grind.dart (hunk 1 above)
Comment (lexaknyazev) [reply tool/grind.dart:65]: The invalid URI error message consists of two lines, so prepending `\n` makes sense there although the layout of that output format makes sense only with monospace fonts.

In any case, ensuring that the table does not contain extra line breaks (even if there aren't any) is an improvement.

**Comment by lexaknyazev** [comment]
> I tried to reproduce issue https://github.com/KhronosGroup/glTF-Validator/issues/180, but this apparently _only_ happens in the web version?

The crash happened because the web version limits the number of validation messages to 16384 and the GLB parser didn't have the proper exception handler for that. The crash should be reproducible with the desktop tool when given a config file. Adding integration tests is planned but not as a part of this release.

**Comment by emackey** [comment]
I haven't had a chance to test this one with my usual tests, but it looks like there are no major changes here, so I'm not expecting any problems from this.  Thanks!

**Pull request merged**